deps: explicitely ask for make generator where needed

LebedevRI opened the pull request:
While https://github.com/JuliaLang/julia/pull/54538 previously
attempted to properly handle this the other way,
by calling `cmake --build`, but that was reverted
with some weird issues on CI,
see https://github.com/JuliaLang/julia/issues/55284

Maybe those issues have gone away by themselves,
maybe not. But let's at least make things work first.

giordano reviewed (changes requested):
#55285, which reverted #54538, which you may be familiar with

LebedevRI committed:
deps: explicitely ask for make generator where needed

While https://github.com/JuliaLang/julia/pull/54538 previously
attempted to properly handle this the other way,
by calling `cmake --build`, but that was reverted
with some weird issues on CI,
see https://github.com/JuliaLang/julia/issues/55284

Maybe those issues have gone away by themselves,
maybe not. But let's at least make things work first.

LebedevRI commented:
Ok, https://github.com/JuliaLang/julia/issues/55284 is unfortunate.
I have not seen it, nor was i tagged in it.
The jobserver mis-iteraction does sound likely,
and i'm not sure how to deal with that.
Was this reported to CMake issue tracker? It really should be.

Zentrik commented:
I didn't report the issue to cmake and I doubt anyone else did.

Zentrik reviewed (approved):
Approved assuming this doesnt actually change anything

LebedevRI committed:
Merge branch 'master' into deps-cmake

LebedevRI commented:
Yes, this PR should be no-op for default configuration, but unbreak non-default ones.

> I didn't report the issue to cmake and I doubt anyone else did.

That's not great either. I don't suppose there is interest
to just try to revert the revert and see if things still go wrong?
My only guess would be to try running cmake this way:
`MAKE=$(MAKE) cmake <...>`  but i'm not sure if it actually reads that env variable.

LebedevRI commented:
> My only guess would be to try running cmake this way:
> `MAKE=$(MAKE) cmake <...>` but i'm not sure if it actually reads that env variable.

Hm, it does, maybe that would solve it. But i'd like to get this PR merged first, please.
```
$ MAKE="/bin/false" cmake -S . -B /tmp/qqq -G"Unix Makefiles"
-- The CXX compiler identification is GNU 14.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - broken
CMake Error at /usr/share/cmake-3.31/Modules/CMakeTestCXXCompiler.cmake:73 (message):
  The C++ compiler

    "/usr/bin/c++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/tmp/qqq/CMakeFiles/CMakeScratch/TryCompile-0PIUT7'
    
    Run Build Command(s): /usr/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_494cc/fast
    /bin/false  -f CMakeFiles/cmTC_494cc.dir/build.make CMakeFiles/cmTC_494cc.dir/build
    gmake: *** [Makefile:134: cmTC_494cc/fast] Error 1
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:18 (project)


-- Configuring incomplete, errors occurred!
```

Zentrik commented:
It you want to test out that change I think you should create a pr similar to https://github.com/JuliaCI/julia-buildkite/pull/443.

LebedevRI commented:
> It you want to test out that change I think you should create a pr similar to [JuliaCI/julia-buildkite#443](https://github.com/JuliaCI/julia-buildkite/pull/443).

https://github.com/JuliaCI/julia-buildkite/pull/447

LebedevRI committed:
Merge branch 'master' into deps-cmake

LebedevRI commented:
@Zentrik @giordano is this sufficient or do you want for this to be run through https://github.com/JuliaCI/julia-buildkite ?

Zentrik reviewed (approved):
Looks the same as last approval so lgtm

LebedevRI commented:
@giordano @Zentrik Thank you!
